worker/include/RTC/TransportTuple.hpp

```diff
@@ -125,6 +130,28 @@ namespace RTC
 				return this->tcpConnection->GetSentBytes();
 		}
 
+	private:
+		void GenerateHash()
+		{
+			const std::string protocol = this->protocol == Protocol::UDP ? "udp" : "tcp";
+			int family;
+			std::string localIp;
+			uint16_t localPort;
+			std::string remoteIp;
+			uint16_t remotePort;
+
+			Utils::IP::GetAddressInfo(GetLocalAddress(), family, localIp, localPort);
+			Utils::IP::GetAddressInfo(GetRemoteAddress(), family, remoteIp, remotePort);
+
+			std::string id = remoteIp + std::to_string(remotePort) + localIp + std::to_string(localPort) +
+			                 std::to_string(family) + protocol;
+
+			this->hash = std::hash<std::string>{}(id);
```

Comment:
Does it need to be `std::string` or can be something non-heap allocated?
Also why not comparing socket addresses directly alongside with protocol?

Reply:
We need a `key` to be used in C++ maps. In `WebRtcServer.hpp`:

```c++
// Map of WebRtcTransports indexed by local ICE usernameFragment.
absl::flat_hash_map<std::string, RTC::WebRtcTransport*> mapLocalIceUsernameFragmentWebRtcTransport;
// Map of WebRtcTransports indexed by TransportTuple.hash.
absl::flat_hash_map<size_t, RTC::WebRtcTransport*> mapTupleWebRtcTransport;
```

This is not about comparing tuples but about having an unique key/hash/id for each tuple so we can use `map.find(key)`.

Note that `TransportTuple.hash` is a `size_t`. Yes, it would be nicer if it could avoid having to generate an intermediate string but we don't know how to do it. @jmillan tried it hard and gave up.

To clarify: given a `TransportTuple` (which has protocol "udp" or "tcp", IP family, local IP, local port, remote IP and remote port) we have to generate a number that it's unique and deterministic for any `TransportTuple` with same internal information.